cf/delete-sample-file-lambda.yaml

```diff
@@ -0,0 +1,141 @@
+                Resource:
+                  - Fn::Sub: "arn:aws:s3:::biomage-originals-${Environment}-${AWS::AccountId}/*"
+        - PolicyName: !Sub "can-create-log-group-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogGroup"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/delete-sample-file-lambda-${Environment}"
+        - PolicyName: !Sub "can-add-logs-in-cloudwatch-${Environment}"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              - Effect: Allow
+                Action:
+                  - "logs:CreateLogStream"
+                  - "logs:PutLogEvents"
+                Resource:
+                  - Fn::Sub: "arn:aws:logs:${AWS::Region}:${AWS::AccountId}:log-group:/aws/lambda/delete-sample-file-lambda-${Environment}:*"
+
+  DeleteSampleFileLambda:
+    Type: AWS::Lambda::Function
+    Properties:
+      FunctionName: !Sub delete-sample-file-lambda-${Environment}
+      Runtime: nodejs14.x
+      Role: !GetAtt DeleteSampleFileLambdaRole.Arn
+      Handler: index.handler
+      Environment:
+        Variables:
+          AWS_ACCOUNT_ID: !Sub "${AWS::AccountId}"
+          CLUSTER_ENV: !Ref Environment
+      Code:
+        ZipFile: |
+          const AWS = require('aws-sdk');
+
+          const deleteSampleFile = (s3_path, bucketName) => {
+            const params = {
+              Bucket: bucketName,
+              Key: s3_path,
+            };
+
+            const s3 = new AWS.S3({
+              apiVersion: '2006-03-01',
+            });
+
+            s3.deleteObject(params, (err) => {
+              if (err) {
+                console.log(err, err.stack);
+                return;
+              }
+              console.log(`== Successfully deleted ${s3_path} ==`);
+            });
+          };
+
+          exports.handler = async (event, context, callback) => {
+            const { s3_path } = event;
+
+            const bucketName = `biomage-originals-${process.env.CLUSTER_ENV}-${process.env.AWS_ACCOUNT_ID}`;
+
+            console.log(`== Deleting ${s3_path} from ${bucketName} ==`);
+            deleteSampleFile(s3_path, bucketName);
+
+            callback(null, event);
+          };
+
+
+  LoggingGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      LogGroupName: !Sub "/aws/lambda/delete-sample-file-lambda-${Environment}"
+      RetentionInDays: 1
+
+
+  DeleteSampleFileLambdaSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription: Allow RDS to connect to lambda
+      VpcId:
+        Fn::ImportValue:
+          !Sub 'eksctl-biomage-${Environment}-cluster::VPC'
+      SecurityGroupEgress:
+        - IpProtocol: tcp
+          FromPort: 443
+          ToPort: 443
+          CidrIp: 0.0.0.0/0
```

Comment:
This seems a more general thing, especially if we are planning to add more lambdas to cleanup other buckets, do you think we could maybe split up the lambda permissions part into a different file and keep it generic so we can reuse it?

Reply:
Hmm... yeap, sure.